### PR TITLE
Assembly robots should use the workflow client provided by dor-services

### PR DESCRIPTION
**Pull request by cbeer**

### CODE REVIEW

**Comment by drh-stanford** [comment]
Can you please explain the context for this PR? Doesn't lyber-core 4.0.3 already handle the changes needed for the workflow service client? https://github.com/sul-dlss/lyber-core/blob/v4.0.3/lib/lyber_core/robot.rb#L57

**Comment by drh-stanford** [comment]
I read through https://github.com/sul-dlss/common-accessioning/pull/53 and I'm understanding this PR now.

File: robots/assembly/base.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+          Dor::Config.workflow.client
+        end
+
+        protected
+
+        def with_item(druid, items_only: Dor::Config.configure.assembly.items_only, &block)
+          ai = item(druid)
+
+          if (items_only && !ai.is_item?)
+             logger.warn("Skipping exif-collect for #{druid} since it is not an item")
```
Comment (drh-stanford) [inline robots/assembly/base.rb]: this warning message is hardcoded for `exif-collect` but as I read it, `jp2-create` and `checksum-compute` need to be handled here, right? maybe use the class name for the warning message?

Comment (cbeer) [inline robots/assembly/base.rb]: Right you are, thanks for catching that.

**Pull request merged**